### PR TITLE
Update batch tally sheet generation to support multiple contests

### PR DESCRIPTION
**Pull request by arsalansufi**
## Overview

Issue link: https://github.com/votingworks/arlo/issues/1815

This PR updates batch tally sheet generation to support multiple contests. Note that the batch audit creation UI still doesn't support creating audits with multiple contests.

I've updated the physical tally sheets in a way that mirrors the recently updated digital tally entry UI (https://github.com/votingworks/arlo/pull/1878). See the new test snapshots for examples of edge case behavior (e.g., two contests with many choices, more than two contests, etc.).

<table>
two-contests
</table>

## Testing

- [x] Updated automated tests
- [x] Cherry-picked changes from my proof-of-concept branch to test manally

### CODE REVIEW

File: client/src/components/JurisdictionAdmin/generateSheets.ts [hunk 1]
```diff
@@ -423,38 +427,52 @@ export const downloadBatchTallySheets = async (
...
+        body: contest.choices.map(choice => [
+          choice.name,
+          '', // Stack totals left blank for the audit board to fill out
+        ]),
+        startY: y,
+        margin: {
+          bottom: pageMargin,
+          left: pageMargin,
+          right: pageMargin,
+          top: pageMargin,
```
Comment (arsalansufi) [inline client/src/components/JurisdictionAdmin/generateSheets.ts:446]: Other than placing the `autoTable` call in a loop, the one other change that I've made here is the addition of this top margin. This makes sure that, if a table begins on a second page, it respects the second page's top margin. Before this PR, I'd been assuming that tables would always be wholly contained on a first page.

File: client/src/components/JurisdictionAdmin/generateSheets.ts [hunk 2]
```diff
@@ -423,38 +427,52 @@ export const downloadBatchTallySheets = async (
...
-      },
-      headStyles: {
-        fontStyle: 'bold',
-      },
-    })
+    for (const [contestIndex, contest] of contests.entries()) {
+      // autoTable automatically adds page breaks
+      autoTable(doc, {
+        head: [
+          [{ content: contest.name, colSpan: 2 }],
```
Comment (jonahkagan) [inline client/src/components/JurisdictionAdmin/generateSheets.ts:434]: A light gray background on the contest name header row might help make it more readable

File: client/src/components/JurisdictionAdmin/generateSheets.ts
Comment (jonahkagan) [inline client/src/components/JurisdictionAdmin/generateSheets.ts:1]: If there's an easy way to keep the check in/check out station steps section all together on the same page, that might be helpful. I can easily see someone forgetting to do one of the steps if it wraps to a different page by itself.

**Review by jonahkagan — APPROVED** [review]

**Review by arsalansufi — COMMENTED** [review]
Oops, forgot to actually submit this comment

File: client/src/components/JurisdictionAdmin/generateSheets.ts
Comment (arsalansufi) [reply client/src/components/JurisdictionAdmin/generateSheets.ts:1]: Good call - _should_ be easy!

Comment (arsalansufi) [reply client/src/components/JurisdictionAdmin/generateSheets.ts:1]: https://github.com/votingworks/arlo/pull/1879/commits/ecfcebbcc0a4d52474d6e6ebeac22ae9d2e06ec8

File: client/src/components/JurisdictionAdmin/generateSheets.ts (hunk 2 above)
Comment (arsalansufi) [reply client/src/components/JurisdictionAdmin/generateSheets.ts:436]: https://github.com/votingworks/arlo/pull/1879/commits/ecfcebbcc0a4d52474d6e6ebeac22ae9d2e06ec8

**Comment by arsalansufi** [comment]
✅ Updated the PR description screenshot to reflect the latest changes

**Comment by jonahkagan** [comment]
Nice, looks great!
